### DISCUSSION TITLE
Please stop the timer when the user is away from the keyboard

### DISCUSSION THREAD

**Discussion by zartc** [monkeytypegame/monkeytype#2021] [Ideas] [answered]
Why is the application still counting time, and lowering the average speed down to zero, when the user is obviously not typing anymore ?

- After a short delay without new keydown recorded the application could reasonably decide to stop counting time.
- When the ESC key is typed the application could without a doubt stop counting time for the obvious reason that the user is not in a typing dreal anymore.

I have a couple of ridiculously low entries in my history because I had to get away from the keyboard for some time while in the middle of a dreal. Nothing told me that the entire duration of the dreal, including dead time, would count in the speed calcultation. On the countrary, the _principle of least astonishement_ would have let me think that the program would have "_paused_" while I was away from the keyboard.

Would it be difficult to implement a mechanism to detect away time that pause the program ?

**Comment by Miodec** [comment, marked as answer]
Pausing has been suggested and refused many times. I don't see a point in it, and it creates issues with calculation, creates questions when it comes to challenges, validity of results when it comes to leaderboards and more. 

Monkeytype was built from the beginning to measure continuous typing speed, and adding pausing is not something I'm planning to add. Sorry.

**Reply by zartc** [reply]
Understood. A least I learned from that: If you can not complete it, abort it! :-)
